storage/rocksdb/ha_rocksdb.cc

```diff
@@ -4164,6 +4196,52 @@ bool ha_rocksdb::init_with_fields() {
   DBUG_RETURN(false); /* Ok */
 }
 
+
+/*
+  If the key is a TTL key, we may need to filter it out.
+
+  The purpose of read filtering for tables with TTL is to ensure that
+  during a transaction a key which has expired already but not removed by
+  compaction yet is not returned to the user.
+
+  Without this the user might be hit with problems such as disappearing
+  rows within a transaction, etc, because the compaction filter ignores
+  snapshots when filtering keys.
+*/
+int ha_rocksdb::should_hide_ttl_rec(const rocksdb::Slice &ttl_rec_val,
+                                    const int64_t curr_ts) const {
+  DBUG_ASSERT(m_pk_descr != nullptr);
+  DBUG_ASSERT(m_pk_descr->has_ttl());
+
+  if (!rdb_is_ttl_read_filtering_enabled() || !rdb_is_ttl_enabled()) {
+    return false;
+  }
+
+  Rdb_string_reader reader(&ttl_rec_val);
+
+  uint64 ts;
+  if (reader.read_uint64(&ts)) {
```

Comment:
This should be a common function that figure out how to read the TTL field from the value based on fields present in the keydef. Right now, it assumes TTL is the first 8 bytes, but that might not always be the case. Consolidating this in one function makes it easy for us to add fields to be read later on.